### DISCUSSION TITLE
Missing go.sum entries?

### DISCUSSION THREAD

**Discussion by mfeilen** [ent/ent#2323] [Q&A]
I have problems with an existing project upgrading to ent@v0.10.0 with Go 1.17:

`go generate ./ent` causes:

../../../pkg/mod/entgo.io/ent@v0.10.0/cmd/internal/printer/printer.go:16:2: missing go.sum entry for module providing package github.com/olekukonko/tablewriter (imported by entgo.io/ent/cmd/internal/printer); to add:
	go get entgo.io/ent/cmd/internal/printer@v0.10.0
../../../pkg/mod/entgo.io/ent@v0.10.0/cmd/internal/base/base.go:24:2: missing go.sum entry for module providing package github.com/spf13/cobra (imported by entgo.io/ent/cmd/ent); to add:
	go get entgo.io/ent/cmd/ent@v0.10.0
../../../pkg/mod/entgo.io/ent@v0.10.0/entc/load/load.go:26:2: missing go.sum entry for module providing package golang.org/x/tools/go/packages (imported by entgo.io/ent/cmd/internal/base); to add:
	go get entgo.io/ent/cmd/internal/base@v0.10.0
../../../pkg/mod/entgo.io/ent@v0.10.0/entc/gen/graph.go:24:2: missing go.sum entry for module providing package golang.org/x/tools/imports (imported by entgo.io/ent/entc/gen); to add:
	go get entgo.io/ent/entc/gen@v0.10.0
ent/generate.go:3: running "go": exit status 1

Quick check of [go.sum](https://github.com/ent/ent/blob/master/go.sum):
`grep -i 'golang.org/x/tools/go/packages' go.sum` > no result
`grep -i 'golang.org/x/tools/imports' go.sum` > no result
`grep -i 'github.com/olekukonko/tablewriter' go.sum`
github.com/olekukonko/tablewriter v0.0.5 h1:P2Ga83D34wi1o9J6Wh1mRuqd4mF/x/lgBS7N7AbDhec=
github.com/olekukonko/tablewriter v0.0.5/go.mod h1:hPp6KlRPjbx+hW8ykQs1w3UBbZlj6HuIJcUGPhkA7kY=
`grep -i 'github.com/spf13/cobra' go.sum`:
github.com/spf13/cobra v1.3.0 h1:R7cSvGu+Vv+qX0gW5R/85dx2kmmJT5z5NM8ifdYjdn0=
github.com/spf13/cobra v1.3.0/go.mod h1:BrRVncBjOJa/eUcVVm9CE+oC6as8k+VYr4NY7WCi9V4=

Quick fix with getting the packages works, but `go tidy` will remove them again...

Any suggestions would be great.

Thanks

**Comment by yonidavidson** [comment]
Can you try go generate -mod=mod ./ent

**Reply by mfeilen** [reply]
Same result

**Comment by a8m** [comment]
How did you install ent and what is the generate command your have in ent/generate.go?

**Reply by mfeilen** [reply]
I updated from 0.9.0 with `go get -u` and then used `go generate ./ent`

**Reply by a8m** [reply]
What is your Go version?

**Reply by mfeilen** [reply]
1.17

**Reply by a8m** [reply]
And you ent/generate.go?

**Reply by mfeilen** [reply]
That was it. Seems to be an older one (probably > 0.6.0 or something). I used a newer version of generate.go and it worked. 

Diff:
old:  `//go:generate go run entgo.io/ent/cmd/ent generate ./schema`
new: `//go:generate go run -mod=mod entgo.io/ent/cmd/ent generate ./schema`

After updating it worked even with `go mod tidy` 
Thank you :-)

**Reply by a8m** [reply]
Wonderful! Thanks 😊
